### DISCUSSION TITLE
Slow Query Performance on VPS Development Server in Laravel

### DISCUSSION THREAD

**Discussion by nerdylive123** [planetscale/discussion#495] [Q&A] [answered]
# DEtails
* I am RUnning the SAme VErsion of PHP, PHP MySQL COnnector, and LAravel on both my LOcal PC and VPS. HOwever, 
* I am experiencing significantly slower query performance on the VPS compared to my local PC, even though they are connected to the same database and using the same branch. I have tried different branches, but the issue persists.

* I expected the VPS to be faster in serving the page and executing queries due to its closer proximity to the database and higher internet speed. 
* The VPS only has traffic from me as it is a development/test server. 
* The test records in the database are minimal, with only a few rows (10-20).

## Environment details:
I am using a Laravel, PHP, MySQL setup with Docker, and the database is hosted on PlanetScale. I have attached some profiler screenshots for reference.

**LOCAL_FEW_QUERIES**
![LOCAL_FEW_QUERIES](https://github.com/planetscale/discussion/assets/70305325/61057280-ac1f-4806-8c9e-f65b7707b6f6)
**LOCAL_NO_QUERIES**
![LOCAL_NO_QUERIES](https://github.com/planetscale/discussion/assets/70305325/f6738c16-cc7f-472d-838a-790017385413)
**VPS_NO_QUERIES**
![VPS_NO_QUERIES](https://github.com/planetscale/discussion/assets/70305325/31095395-5cf4-4568-94af-596de27dc006)
**VPS_FEW_QUERIES**
![VPS_FEW_QUERIES](https://github.com/planetscale/discussion/assets/70305325/82dbdc02-1f69-429a-ab2d-235b1b2dd75c)

## DB Schema:
**Users**
![image](https://github.com/planetscale/discussion/assets/70305325/c96a8335-6eee-4de5-b046-2bce268ae520)
**Roles**
![image](https://github.com/planetscale/discussion/assets/70305325/a6f96109-971a-4913-a7d4-b3ab747a2d27)

**Comment by nerdylive123** [comment]
additional info: its hosted on digital ocean, SG with the db on SG too. and with no firewalls on.

**Comment by mscoutermarsh** [comment]
Which PlanetScale region is your database in?

It may be an issue with the routing of the request. Which might be able to be fixed by using the region host in the connection string.

I can provide the host once I know which region.

**Comment by nerdylive123** [comment]
It's in Singapore ( ap southeast 1) and yes I've contacted planetscale's
support and they gave me the region host I tried it and it worked, so I
think I'm gonna use that

**Comment by nerdylive123** [comment]
Does some cloud provider have that or only some? I'm using digital ocean
for this problem

**Reply by mscoutermarsh** [reply]
Great. It's a fairly rare issue that impacts some providers in some regions.

**Reply by nerdylive123** [reply]
Oh alright then thanks for your help!

**Comment by markmarilag27** [comment]
Hi @nerdylive123 can you please share the `region host` I had similar issue, but I cant find the documentation regarding `region host` thanks in advance!

**Comment by nerdylive123** [comment, marked as answer]
oh hey, same issue with me huh sorry for the long reply.. heres what i found from the support that helped me out
 ## Note: if you're using the **-****-1 you wont need the "aws"-**-***** for example: 
- ap-southeast.connect.psdb.cloud ( SG )
- aws-ap-southeast-2.connect.psdb.cloud ( Sydney )
## The list of region name slugs can be found here:

- AWS ap-northeast-1 (Tokyo) — ap-northeast
- AWS ap-south-1 (Mumbai) — ap-south
- AWS ap-southeast-1 (Singapore) — ap-southeast
- AWS ap-southeast-2 (Sydney) — aws-ap-southeast-2
- AWS eu-central-1 (Frankfurt) — eu-central
- AWS eu-west-1 (Dublin) — eu-west
- AWS eu-west-2 (London) — aws-eu-west-2
- AWS sa-east-1 (Sao Paulo) — aws-sa-east-1
- AWS us-east-1 (Northern Virginia) — us-east
- AWS us-east-2 (Ohio) — aws-us-east-2
- AWS us-west-2 (Oregon) — us-west

[Planetscale regions ( AWS )](https://planetscale.com/docs/concepts/regions#aws-regions)

**Reply by nerdylive123** [reply]
@markmarilag27 ping

**Reply by markmarilag27** [reply]
Awesome! Thank you very much
